segmenters/nlp/NLTKSentencizer/__init__.py

```diff
@@ -0,0 +1,120 @@
+
+
+    :param language: default='english'. Lowercased language name to initialize the sentence tokenizer, accepted languages are listed in :attr:`SUPPORTED_LANGUAGES`.
+    :type language: Optional[str]
+    :param resource_url: default=None. Optional parameter to specify the location for tokenizer model, overrides `language`. For a file path, should follow `file://path_to_resource` format. For more information, see https://www.nltk.org/data.html
+    :type resource_url: Optional[str]
+    :param args: Additional positional arguments
+    :param kwargs: Additional keyword arguments
+    """
+
+    SUPPORTED_LANGUAGES = [
+        'czech',
+        'danish',
+        'dutch',
+        'english',
+        'estonian',
+        'finnish',
+        'french',
+        'german',
+        'greek',
+        'italian',
+        'norwegian',
+        'polish',
+        'portuguese',
+        'russian',
+        'slovene',
+        'spanish',
+        'swedish',
+        'turkish',
+    ]
+
+    def __init__(
+        self,
+        language: Optional[str] = 'english',
+        resource_url: Optional[str] = None,
+        *args,
+        **kwargs,
+    ):
+        """Set constructor"""
+        super().__init__(*args, **kwargs)
+        if resource_url:
+            self.resource_url = resource_url
+            self.language = None
+        else:
+            self.resource_url = f'tokenizers/punkt/{language}.pickle'
+            self.language = language
+
+    def post_init(self):
+        from nltk.data import load
+
+        try:
+            self._sent_tokenizer = load(self.resource_url)
+        except LookupError:
+            if self.language in self.SUPPORTED_LANGUAGES:
+                try:
+                    import nltk
+
+                    nltk.download('punkt')
+                    self._sent_tokenizer = load(self.resource_url)
+                except:
+                    self.logger.error(
+                        f'Please ensure that "nltk_data" folder is accessible to your working directory.'
+                    )
+                    raise
+            elif self.language:
+                self.logger.error(
+                    f'The language you specified ("{self.language}") is not supplied by NLTK.'
+                    f'Please ensure that language is one of the acceptable languages listed in '
+                    f'{self.SUPPORTED_LANGUAGES}.\nFor latest list of supported languages check out '
+                    f'https://raw.githubusercontent.com/nltk/nltk_data/gh-pages/index.xml with '
+                    f'id="punkt"'
+                )
+                raise
+            else:
+                self.logger.error(
+                    f'The resource_url you provided could not be loaded: {self.resource_url}'
+                )
+                raise
+        self.logger.info(f'NLTK {self.language} sentence tokenizer ready.')
+
+    def segment(self, text: str, *args, **kwargs) -> List[Dict]:
```

Comment:
Hey @mezig351 , since the last patch, we need the `segmenters` to be decorated with `single` decorator to make sure it works with the rest of the pipeline. It is not yet documented, but it is needed, you can see some `segmenters` that have been recently merged into master to see how that works